independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/TypeVariableImpl.java

```diff
@@ -54,4 +54,22 @@ public AnnotatedType[] getAnnotatedBounds() {
         throw new UnsupportedOperationException();
     }
 
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(name);
+        if (bounds.size() > 0) {
+            sb.append(" extends ");
+            for (Type bound : bounds) {
+                if (bound instanceof Class) {
+                    sb.append(((Class<?>) bound).getName());
+                } else {
+                    sb.append(bound);
+                }
+                sb.append(" & ");
+            }
+            sb.delete(sb.length() - 3, sb.length());
```

Comment:
Why don't you use `Iterator` here? This feels "unnecessarily hacky"...

Reply:
I copied this from `ParameterizedTypeImpl` if I remember correctly :-)

(My personal solution is to have a boolean variable `first` that starts its life as `true` but is set to `false` ad the end of each iteration, but I just couldn't bother here.)

That said, it would probably be best to use `StringJoiner` here.

Reply:
Ok, `ParameterizedTypeImpl` was probably copied from Weld ;-)

Reply:
So I have an implementation with `StringJoiner` that looks a bit simpler:

```java
@Override
public String toString() {
    StringJoiner joiner = new StringJoiner(" & ", " extends ", "");
    joiner.setEmptyValue("");
    for (Type bound : bounds) {
        if (bound instanceof Class) {
            joiner.add(((Class<?>) bound).getName());
        } else {
            joiner.add(bound.toString());
        }
    }
    return name + joiner;
}
```

Obviously, there's a bit more overhead. I'm fine with both, so your choice :-)

Reply:
`StringJoiner` looks ok.